docs/dg/dev/architecture/dependency-injection.md

```diff
@@ -0,0 +1,267 @@
+
+To compile the container, you have two options:
+
+- On every request to the application, the Kernel checks if a compiled container already exists. If not, or if the cache is outdated, it generates a new one. (perfect in dev mode)
+- Run the following command to build the cache. The application then immediately uses the freshly compiled container.
+
+```bash
+console container:build
+```
+
+## How Dependency Injection is triggered
+
+Dependency injection is activated when a service is first requested from the container. For instance, if a controller is defined as a service in the container, the container instantiates it. If this controller's constructor has arguments (dependencies), the container automatically injects the required services. This process continues down the dependency chain.
+
+If a class is not defined as a service in the container, it will be instantiated the classic Spryker way, and no dependencies will be injected.
+
+## Using services
+
+Once the container is compiled, you can use your services in your application. You can inject them into your controllers, and other services via constructor injection.
+
+### Controller example
+
+Here is an example of how to inject a service into a controller.
+
+First, define your controller and inject the service using constructor property promotion:
+
+`src/Pyz/Yves/MyModule/Controller/IndexController.php`
+
+```php
+<?php
+
+namespace Pyz\Yves\MyModule\Controller;
+
+use Spryker\Yves\Kernel\Controller\AbstractController;
+use Pyz\Yves\MyModule\MyService;
+
+class IndexController extends AbstractController
+{
+    public function __construct(private MyService $myService)
+    {
+    }
+
+    public function indexAction()
+    {
+        $data = $this->myService->getSomeData();
+
+        return $this->view(
+            '@MyModule/views/index/index.twig',
+            ['data' => $data]
+        );
+    }
+}
+```
+
+Because we configured service discovery for the `Pyz\Yves\MyModule` namespace in `config/ApplicationServices.php` and autowiring is enabled, MyService will be available in the Container when requested.
+
+
+### Model/Service example
+
+Here is an example of a service with a dependency on a repository.
+
+`src/Pyz/Yves/MyModule/SomeService.php`
+
+```php
+<?php
+
+namespace Pyz\Yves\MyModule;
+
+class SomeService
+{
+    public function findData(): array
+    {
+        // some logic to fetch data
+        return ['foo' => 'bar'];
+    }
+}
+```
+
+`src/Pyz/Yves/MyModule/MyService.php`
+
+```php
+<?php
+
+namespace Pyz\Yves\MyModule;
+
+class MyService
+{
+    public function __construct(private SomeService $someService)
+    {
+    }
+
+    public function getSomeData(): array
+    {
+        return $this->someService->findData();
+    }
+}
+```
+
+Because we configured service discovery for the `Pyz\Yves\MyModule` namespace in `config/Symfony/{APPLICATION}/ApplicationServices.php` and autowiring is enabled, `SomeService` is automatically injected into `MyService`.
```

Comment:
Maybe:
```suggestion
Because we configured service discovery for the `Pyz\Yves\MyModule` namespace in `config/Symfony/Yves/ApplicationServices.php` and autowiring is enabled, `SomeService` is automatically injected into `MyService`.
```